pyega3/data_file.py

```diff
@@ -257,17 +275,15 @@ def download_file_retry(self, num_connections, output_file, genomic_range_args,
                 logging.exception(e)
                 if num_retries == max_retries:
                     if DataFile.temporary_files_should_be_deleted:
-                        self.delete_temporary_files()
+                        self.delete_temporary_folder(temporary_directory)
 
                     raise e
                 time.sleep(retry_wait)
                 num_retries += 1
                 logging.info(f"retry attempt {num_retries}")
 
-    def delete_temporary_files(self):
+    def delete_temporary_folder(self, temporary_directory):
         try:
-            for temporary_file in self.temporary_files:
-                logging.debug(f'Deleting the {temporary_file} temporary file...')
-                os.remove(temporary_file)
+            shutil.rmtree(temporary_directory)
```

Comment:
I presume/hope that this works even when the temporary_directory/output_file has strange characters (space, Chinese chars, a-character-which-Ashutosh-fixed-the-last-time) in its name.

Reply:
https://bugs.python.org/issue24672 says that it used to be a problem in python 2, but is not a problem in python 3 where all strings are unicode.